webdriver/tests/bidi/browsing_context/set_viewport/device_pixel_ratio.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from ... import get_device_pixel_ratio, get_viewport_dimensions
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("device_pixel_ratio", [1, 2])
+async def test_device_pixel_ratio_only(bidi_session, new_tab, device_pixel_ratio):
+    viewport = await get_viewport_dimensions(bidi_session, new_tab)
+
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        device_pixel_ratio=device_pixel_ratio)
+
+    assert await get_device_pixel_ratio(bidi_session, new_tab) == device_pixel_ratio
+    assert await get_viewport_dimensions(bidi_session, new_tab) == viewport
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("device_pixel_ratio", [1, 2])
+async def test_device_pixel_ratio_with_viewport(bidi_session, new_tab, device_pixel_ratio):
+    test_viewport = {"width": 250, "height": 300}
```

Comment:
nit: we do not know the original viewport. Maybe use first fetch the current one and set it to the half width and height? Alternatively we should assert here that the values are different.

Reply:
Adding an assertion.